src/vma/dev/net_device_table_mgr.cpp

```diff
@@ -230,6 +230,19 @@ int net_device_table_mgr::map_net_devices()
 			continue;
 		}
 
+#ifdef DEFINED_VMAPOLL
+		// only support mlx5 device for vmapoll
+		if(strncmp(ib_ctx->get_ibv_device()->name, "mlx5", 4) != 0) {
```

Comment:
May be more directly to check against mlx4 as
`strncmp(ib_ctx->get_ibv_device()->name, "mlx4", 4)== 0`